kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/convert/BigQueryRecordConverter.java

```diff
@@ -73,13 +67,53 @@ public BigQueryRecordConverter(boolean shouldConvertDoubleSpecial) {
    */
   public Map<String, Object> convertRecord(SinkRecord kafkaConnectRecord) {
     Schema kafkaConnectSchema = kafkaConnectRecord.valueSchema();
+    Object kafkaConnectValue = kafkaConnectRecord.value();
+    if (kafkaConnectSchema == null) {
+      if (kafkaConnectValue instanceof Map) {
+        return (Map<String, Object>) validateSchemalessRecord(kafkaConnectValue);
+      }
+      throw new ConversionConnectException("Only Map objects supported in absence of schema for " +
+              "record conversion to BigQuery format.");
+    }
     if (kafkaConnectSchema.type() != Schema.Type.STRUCT) {
       throw new
           ConversionConnectException("Top-level Kafka Connect schema must be of type 'struct'");
     }
     return convertStruct(kafkaConnectRecord.value(), kafkaConnectSchema);
   }
 
+  private Object validateSchemalessRecord(Object value) {
+    if (ClassUtils.isPrimitiveWrapper(value.getClass()) || value instanceof String) {
+      return value;
+    }
+    if (value instanceof byte[] || value instanceof ByteBuffer) {
+      return convertBytes(value);
+    }
+    if (value instanceof ArrayList) {
+      return
+          ((ArrayList) value).stream().map(
+                  v -> validateSchemalessRecord(v)
+          ).collect(Collectors.toList());
+    }
+    if (value instanceof Map) {
+      return
+        ((Map<?, ?>) value).entrySet().stream().collect(
+                Collectors.toMap(
+                        entry -> {
+                          if (!(entry.getKey() instanceof String)) {
+                            throw new DataException(
+                                    "Map objects in absence of schema needs to have string value keys.");
+                          }
+                          return entry.getKey();
+                        },
+                        entry -> validateSchemalessRecord(entry.getValue())
+                )
+        );
+    }
+    throw new ConversionConnectException("Invalid data type found in schemaless record. " +
```

Comment:
"Unsupported" rather than "Invalid" would be more informative. We should also add information about the type we encountered to help the user debug those exceptions.